pkg/client/restclient/request.go

```diff
@@ -800,6 +806,7 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 	// TODO: Change to a timeout based approach.
 	maxRetries := 10
 	retries := 0
+	initialUrl := r.urlProvider.Get()
 	for {
 		url := r.URL().String()
```

Comment:
Are we calling `r.URL()` here multiple times and expecting the same value?  If so, we should really stop doing that and return it once.